src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,47 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+public class FindCommand extends Command {
+
+    private String keyword;
+
+    /**
+     * Initialises a FindCommand.
+     *
+     * @param keyword The keyword to find in tasks.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword.toLowerCase();
+    }
+
+    /**
+     * Executes the Find Command and lists all existing tasks in the TaskList that contains the keyword.
+     *
+     * @param tasks   The TaskList containing all existing tasks.
+     * @param ui      The Ui for printing purposes.
+     * @param storage The Storage for saving tasks to file.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+
+        // Print all existing items in the list
+        ui.printMessage("Here are the matching tasks in your list:");
+        for (Task x : tasks.allTasks) {
+            if (x.toString().toLowerCase().contains(keyword)) {
```

Comment:
Similar to `DeleteCommand`, you may consider making it more OOP by encapsulating the finding process within the `TaskList` class.